client/state/google-apps-users/reducer.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * External Dependencies
+ */
+
+import { combineReducers } from 'redux';
+import uniqBy from 'lodash/uniqBy';
+import property from 'lodash/property';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	GOOGLE_APPS_USERS_FETCH,
+	GOOGLE_APPS_USERS_FETCH_COMPLETED,
+	GOOGLE_APPS_USERS_FETCH_FAILED,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+
+export function items( state = [], action ) {
+	switch ( action.type ) {
+		case GOOGLE_APPS_USERS_FETCH_COMPLETED:
+			return uniqBy( state.concat( action.items ), property( 'email' ) );
+		case SERIALIZE:
+			return [];
+		case DESERIALIZE:
+			return [];
+	}
+	return state;
+}
+
+export function loaded( state = false, action ) {
+	switch ( action.type ) {
+		case GOOGLE_APPS_USERS_FETCH:
+		case SERIALIZE:
+		case DESERIALIZE:
+			return false;
```

Comment:
Yes we do re-fetch, if the user changes view from one domain to another, we want them to see a loading indicator instead of an empty list.

Similarly, if the user moves from domain view to sitewide view, they'll temporarily see an incomplete list; I would rather show a loading indicator instead of a broken list.